src/python_testing/TC_SU_2_5.py

```diff
@@ -0,0 +1,509 @@
+            cluster=Clusters.BasicInformation,
+            attribute=Clusters.BasicInformation.Attributes.SoftwareVersion,
+            node_id=self.requestor_node_id)
+        update_state_attr_handler = AttributeSubscriptionHandler(
+            expected_cluster=Clusters.OtaSoftwareUpdateRequestor,
+            expected_attribute=Clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState
+        )
+        await update_state_attr_handler.start(dev_ctrl=self.controller, node_id=self.requestor_node_id, endpoint=0,
+                                              fabric_filtered=False, min_interval_sec=0, max_interval_sec=5)
+        await self.announce_ota_provider(self.controller, self.provider_node_id, self.requestor_node_id)
+
+        # Wait Until Downlading
+        update_state_match = AttributeMatcher.from_callable(
+            "Waiting Update state is Downloading",
+            lambda report: report.value == Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kDownloading)
+
+        update_state_attr_handler.await_all_expected_report_matches([update_state_match], timeout_sec=90)
+        # Wait some time to check if is downloading
+        await asyncio.sleep(5)
+        # Verify the default download path and the file size
+        # Read file for /tmp/test.bin should exists and greater than 0
+        ota_file_data = self.get_downloaded_ota_image_info()
+        logger.info(f"Downloaded ota image data {str(ota_file_data)}")
+        asserts.assert_equal(True, ota_file_data['exists'], f"File is was not downloaded  at {ota_file_data['path']}")
+        asserts.assert_greater(ota_file_data['size'], 0, "Downloaded file is still at 0")
+
+        # Applying
+        update_state_match = AttributeMatcher.from_callable(
+            "Waiting Update state is Applying",
+            lambda report: report.value == Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kApplying)
+
+        update_state_attr_handler.await_all_expected_report_matches(
+            [update_state_match], timeout_sec=self.ota_image_download_timeout)
+        update_state_match = AttributeMatcher.from_callable(
+            "Waiting Update state is Idle",
+            lambda report: report.value == Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kIdle)
+
+        update_state_attr_handler.await_all_expected_report_matches([update_state_match], timeout_sec=60)
+
+        update_state_attr_handler.reset()
+        await update_state_attr_handler.cancel()
+
+        # Make sure attr is Idle
+        update_state = await self.read_single_attribute_check_success(
+            Clusters.OtaSoftwareUpdateRequestor, Clusters.OtaSoftwareUpdateRequestor.Attributes.UpdateState, self.controller, self.requestor_node_id, 0)
+        asserts.assert_equal(update_state, Clusters.OtaSoftwareUpdateRequestor.Enums.UpdateStateEnum.kIdle,
+                             "Update state is not idle")
+        ota_file_data = self.get_downloaded_ota_image_info()
```

Comment:
This was checked above...does this change? is the download guaranteed to be complete before the earlier check?